tests/Operation/AggregateFunctionalTest.php

```diff
@@ -109,6 +110,46 @@ public function testTypeMapOption(array $typeMap = null, array $expectedDocument
         $this->assertEquals($expectedDocuments, $results);
     }
 
+    public function testExplainOption()
+    {
+        $this->createFixtures(3);
+
+        $pipeline = [['$match' => ['_id' => ['$ne' => 2]]]];
+        $operation = new Aggregate($this->getDatabaseName(), $this->getCollectionName(), $pipeline, ['explain' => true, 'typeMap' => ['root' => 'array']]);
+        $results = iterator_to_array($operation->execute($this->getPrimaryServer()));
+
+        $this->assertCount(1, $results);
+        $this->assertArrayHasKey('stages', $results[0]);
+    }
+
+    public function testExplainOptionWithWriteConcern()
+    {
+        if (version_compare($this->getServerVersion(), '3.4.0', '<')) {
+            $this->markTestSkipped('Sessions are not supported');
+        }
+
+        $this->createFixtures(3);
+
+        $pipeline = [['$match' => ['_id' => ['$ne' => 2]]], ['$out' => $this->getCollectionName() . '.output']];
+        $options = ['explain' => true, 'writeConcern' => new WriteConcern(1)];
+
+        (new CommandObserver)->observe(
+            function() use ($pipeline, $options) {
+                $operation = new Aggregate($this->getDatabaseName(), $this->getCollectionName(), $pipeline, $options);
+
+                $results = iterator_to_array($operation->execute($this->getPrimaryServer()));
+
+                $this->assertCount(1, $results);
+                $this->assertObjectHasAttribute('stages', current($results));
```

Comment:
Did `$this->assertArrayHasKey('stages', $results[0]);` not work here?

I'm actually curious how this works given the implementation of [`ObjectHasAttribute::matches()`](https://github.com/sebastianbergmann/phpunit/blob/master/src/Framework/Constraint/ObjectHasAttribute.php#L30):

```php
$a = new ArrayObject(['foo'=>1]);
$ro = new ReflectionClass($a);
$ro->hasProperty('foo'); // false
```

Reply:
Nevermind, I'm an idiot.

You're using the Aggregation operation class directly, so there is no type map and this is a `stdClass` instance.